Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/submissions/submission-details/SubmissionDetailsPage.tsx

```diff
@@ -121,6 +128,7 @@ const SubmissionDetailsPage = () => {
 
     const renderSolutionDetails = useCallback(() => {
         const { allowBinaryFilesUpload } = submissionType || {};
+        const shouldLoadData = !isRetestingStarted && (testRuns && testRuns.length !== 0);
```

Comment:
Why not load data if there are no test runs?

Reply:
Added a comment which explains what the purpose of the 'shouldLoadData' constant is.

Reply:
In Legacy Judge StartedExecutionOn and CompletedExecutionOn are not shown when there is a compilation error, that is why I hid them when there is a compilation error. For the other case, when the submission is being retested, they are hidden since otherwise the dates of the previous submission's output will be displayed. Again, here in Legacy Judge, they are not displayed, whilst ModifiedOn is, but in our implementation if we show ModifiedOn, it will be will the old submission's date, since the data is not yet updated from the new submission's execution.

Reply:
And if there is Compile time error, how are we going to display the Modified On? There is no problem to display the `StartedExecutionOn` and `CompletedExecutionOn` even in compile time error, as this info is only for admins..

Reply:
```const shouldLoadData = (!isRetestingStarted && (testRuns && testRuns.length !== 0)) || !isCompiledSuccessfully;``` would show the ```StartedExecutionOn```, ```CompletedExecutionOn``` and ```ModifiedOn``` when there is a compilation error. When it comes to retesting a submission, they will not be displayed during the retest, but after it has been completed.